examples/pennylane_QGAN.py

```diff
@@ -0,0 +1,224 @@
+    prob_real_true = (true_disc_output + 1) / 2
+    return prob_real_true
+
+
+def prob_fake_true(gen_weights, disc_weights):
+    fake_disc_output = gen_disc_circuit(gen_weights, disc_weights)
+    # convert to probability
+    prob_fake_true = (fake_disc_output + 1) / 2
+    return prob_fake_true  # generator wants to minimize this prob
+
+
+def disc_cost(disc_weights):
+    cost = prob_fake_true(gen_weights, disc_weights) - prob_real_true(disc_weights)
+    return cost
+
+
+def gen_cost(gen_weights):
+    return -prob_fake_true(gen_weights, disc_weights)
+
+
+##############################################################################
+# Optimization
+# ~~~~~~~~~~~~
+#
+# We initialize the fixed angles of the “real data” circuit, as well as
+# the initial parameters for both generator and discriminator. These are
+# chosen so that the generator initially prepares a state on wire 0 that
+# is very close to the :math:`\left| 1 \right\rangle` state.
+
+phi = np.pi / 6
+theta = np.pi / 2
+omega = np.pi / 7
+np.random.seed(0)
+eps = 1e-2
+gen_weights = np.array([np.pi] + [0] * 8) + np.random.normal(scale=eps, size=[9])
+disc_weights = np.random.normal(size=[9])
+
+##############################################################################
+# We begin by creating the optimizer:
+
+opt = GradientDescentOptimizer(0.1)
+
+##############################################################################
+# In the first stage of training, we optimize the discriminator while
+# keeping the generator parameters fixed.
+
+for it in range(50):
+    disc_weights = opt.step(disc_cost, disc_weights)
+    cost = disc_cost(disc_weights)
+    if it % 5 == 0:
+        print("Step {}: cost = {}".format(it + 1, cost))
+
+##############################################################################
+# At the discriminator’s optimum, the probability for the discriminator to
+# correctly classify the real data should be close to one.
+
+print(prob_real_true(disc_weights))
+
+
+##############################################################################
+# For comparison, we check how the discriminator classifies the
+# generator’s (still unoptimized) fake data:
+
+print(prob_fake_true(gen_weights, disc_weights))
+
+
+##############################################################################
+# In the adverserial game we have to now train the generator to better
+# fool the discriminator (we can continue training the models in an
+# alternating fashion until we reach the optimum point of the two-player
+# adversarial game).
+
+for it in range(200):
+    gen_weights = opt.step(gen_cost, gen_weights)
+    cost = -gen_cost(gen_weights)
+    if it % 5 == 0:
+        print("Step {}: cost = {}".format(it, cost))
+
+
+##############################################################################
+# At the optimum of the generator, the probability for the discriminator
+# to be fooled should be close to 1.
+
+print(prob_real_true(disc_weights))
```

Comment:
```suggestion
print(prob_fake_true(disc_weights))
```
This is a typo that's been lurking in there for a long time :laughing:

Reply:
already taken care of in the PRs on tutorials.